mu-plugins/blocks/ratings-bars/render.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+$attributes = apply_filters( 'wporg_block_ratings_bars', $attributes );
```

Comment:
```suggestion
$attributes = apply_filters( 'wporg_ratings_bars_attributes', $attributes );
```

The rough naming scheme I've been using is snake-case block name for a prefix (so `wporg/ratings-bars` becomes `wporg_ratings_bars`) and a descriptor of the filter value (in this case, `attributes`).

Is filtering all of the attributes is necessary here? Other things like styles, alignment, etc will come through as well. Do you think we'll ever want to have a static `<!-- wp:wporg/ratings-stars {"rating":80} /-->` block? (other block, but easier to write 🙂 )

I would also give this a doc block following [the hooks doc standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters), you can see an example on [the wporg_query_filter_options_* filter](https://github.com/WordPress/wporg-mu-plugins/blob/652c3585fcc901116263eaf580753d916d6bf3ea/mu-plugins/blocks/query-filter/render.php#L6-L29).

Reply:
The filter should also pass back the post ID from the block context ($block->context['postId']), just to ensure the filtered ratings always pull from the correct post.

Reply:
> Is filtering all of the attributes is necessary here?

I would prefer to only filter the block attributes but I don't see a way to do that without forcing consumers to add multiple filters for each parameter. Is there a different approach I should be considering?

Reply:
> Do you think we'll ever want to have a static <!-- wp:wporg/ratings-stars {"rating":80} /--> block?

I'm not sure. If the answer is no, how would you architect this differently?

Reply:
It seemed easier to show than describe, so my approach is here: https://github.com/WordPress/wporg-mu-plugins/compare/try/rating_blocks...try/rating_blocks-alt

- The filter runs over an empty array, which is populated by the hooked function in the child themes, so it doesn't need to run over the classnames, etc.
- I used `wp_parse_args` to ensure the resulting array had the right formatting, so then we can just check a few things to make sure the filter did provide data.
- It is required to provide the data via filter, if no hooked function returns the data, the block will not render.
- Removed the `slug` arg, and built the support URL in the hooked function, ex `wordpress.org/support/theme/SLUG/reviews/`. This + `add_query_arg` create the URL in the render function. It would be nice to replace this with the block binding, but that's not available to custom blocks yet.
    - I thought about leaving this as an attribute, but think it makes sense to build it dynamically like this.
- Use one filter for both blocks. I don't feel _very_ strongly about this, but they use a similar format, so it seems easier in the child theme to only provide the ratings data once.

Now, the child theme hook would look like this:

```php
use function WordPressdotorg\Theme\Theme_Directory_2024\{get_support_url, get_theme_information};

/**
 * Update ratings blocks with real rating data.
 *
 * @param array $data    Rating data.
 * @param int   $post_id Current post.
 *
 * @return array
 */
function set_rating_data( $data, $post_id ) {
	$theme = get_theme_information( $post_id );
	if ( ! $theme ) {
		return $data;
	}

	return array(
		'rating' => $theme->rating,
		'ratingsCount' => $theme->num_ratings,
		'ratings' => $theme->ratings,
		'supportUrl' => get_support_url( $theme->slug . '/reviews/' ),
	);
}
add_filter( 'wporg_ratings_data', __NAMESPACE__ . '\set_rating_data', 10, 2 );
```

Feel free to use any of that branch's code that you want, but that's my "how I'd do it" answer 🙂

Reply:
I see. Yeah that seems logical to me. I don't have much of a preference here. If you think it's better, we can go with that.